packages/@uppy/status-bar/src/StatusBarUI.tsx

```diff
@@ -199,52 +237,10 @@ export default function StatusBarUI<M extends Meta, B extends Body>({
         aria-valuenow={progressValue!}
       />
 
-      {(() => {
-        switch (uploadState) {
-          case STATE_PREPROCESSING:
-          case STATE_POSTPROCESSING:
-            return (
-              <ProgressBarProcessing
-                progress={calculateProcessingProgress(files)}
-              />
-            )
-          case STATE_COMPLETE:
-            return <ProgressBarComplete i18n={i18n} />
-          case STATE_ERROR:
-            return (
-              <ProgressBarError
-                error={error}
-                i18n={i18n}
-                numUploads={numUploads}
-                complete={complete}
-              />
-            )
-          case STATE_UPLOADING:
-            return (
-              <ProgressBarUploading
-                i18n={i18n}
-                supportsUploadProgress={supportsUploadProgress}
-                totalProgress={totalProgress}
-                showProgressDetails={showProgressDetails}
-                isUploadStarted={isUploadStarted}
-                isAllComplete={isAllComplete}
-                isAllPaused={isAllPaused}
-                newFiles={newFiles}
-                numUploads={numUploads}
-                complete={complete}
-                totalUploadedSize={totalUploadedSize}
-                totalSize={totalSize}
-                totalETA={totalETA}
-                startUpload={startUpload}
-              />
-            )
-          default:
-            return null
-        }
-      })()}
+      {progressBarStateEl}
```

Comment:
I think we can undo this change. Makes the PR more clear and it's not necessarily an improvement.

Reply:
It's not a prettification, we take `progressBarStateEl` up because we need that value in order to determine whether we should collapse the footer.

In normal html, this footer would be automatically collapsed when nothing's inside. In our case, we collapse it manually via the height property (controlled by `aria-hidden`), because we need animations.
My point with this change is to make this `aria-hidden` _derived_.
Nothing is inside (each child determines via some fancy if conditions whether it needs to be here)? Collapse the footer.

Reply:
If you like we can revert this change and write

```js
const thereIsNothingInside = !atLeastOneAction && uploadState === STATE_WAITING
```

(logically equivalent, but `aria-hidden` stops being derived from the presence of children).

I like this solution too, was on the brim here.